includes/abstracts/abstract-wc-data.php

```diff
@@ -28,6 +28,18 @@
 	protected $data = array();
 
 	/**
+	 * Core data changes for this object.
+	 * @var array
+	 */
+	protected $changes = array();
+
+	/**
+	 * This is false until the object is read from the DB.
+	 * @var bool
+	 */
+	protected $object_read = false;
```

Comment:
If the data stores are going to be responsible for doing the actual reading, we need to have a way to set the flag from somewhere else too.